### ISSUE TITLE
Protean (SA: RET11) - compliance checks

### ISSUE THREAD

**Issue by sandeepshahi** [ONDC-Official/v1.2.0-logs#2231]
## Flow 1

**/on_search**

- id in bpp/providers/categories should be alphanumeric and upto 12 characters
- upper price in price/tags/range/upper seems to be calculated incorrectly; eg: for base item - "ec327aef-5d85-4195-8991-ec4b9f6bc84e", shouldn't the price be (150)+(160)+(50+60)

**/on_search (inc)**

- txn id should be the same as in /search (inc)

## Flow 2

**/on_select**

- tracking must be supported for "Immediate Delivery" (F&B) orders

**/on_init**

- inconsistency in the attribute value /fulfillments/end/contact/email; getting removed and added irregularly in the subsequent APIs

**/on_status (pending)**

- start of pickup time range can't be context/timestamp;check /fulfillments/start/time/range/start

## Flow 3

**/on_select (OOS)**

- invalid JSON path of error object; should be outside message block

## Flow 6

**/on_confirm**

- how can estimated pickup/delivery time ranges be provided even before the order is accepted? weren't present in previous flows either

@RaviV-Protean

**Comment by RaviV-Protean** [comment]
@sandeepshahi, Thanks for your feedback. 

1. Will resubmit flow 1. 
2. Will resubmit flow 2. 
3. Will resubmit flow 3 with select (oos) and on_select (oos)
4. As per our understanding and documentation we read sellers can edit the order before Accepting it, so there should not be any issue in that.

**Comment by RaviV-Protean** [comment]
@sandeepshahi, PR has been raised. https://github.com/ONDC-Official/v1.2.0-logs/pull/2234

**Comment by sandeepshahi** [comment]
@RaviV-Protean, have you fixed flow 6 issues? There were discrepancies in the implementation compared to the other flows.

**Comment by RaviV-Protean** [comment]
@sandeepshahi, Yes we have rectified and resolved at our end. Will be providing pickup/delivery ranges once the order is accepted.

**Comment by sandeepshahi** [comment]
## RET11

**/on_search**

- lower price in price/tags/range/upper seems to be calculated incorrectly; eg: for base item - "b83a7cb6-d195-4455-99c2-120e417bdc53", shouldn't the price be 160 (base item) + 120 (mandatory customization)

## Flow 2

- pickup and delivery contact details are the same; Make sure right info is being captured and sent back
- Make sure the invoice is generated correctly in prod

## Flow 3

**/on_select (OOS)**

- invalid JSON path of error object; should be outside message block


@RaviV-Protean

**Comment by RaviV-Protean** [comment]
@sandeepshahi, Thanks for your feedback.

1. Mandatory customizations for item.id - "b83a7cb6-d195-4455-99c2-120e417bdc53" is "124da4ad-3be5-4d0d-a302-28c031b18d3e" and price of that is 160 (base item) + 170 (mandatory customization) = 330 which is defined in price.

2. As we are testing we have kept both details same (on seller side as well as buyer application) - It's being captured correctly. 

3. Yes, it will get generated correctly in production for sure. 

4. Got it, That's at our end. we have tested in log-validation-utility as well. It was not getting flagged using that as well, it's very minor fix at our end. 

@sandeepshahi, Can we get clearance for tech logs and move to QA-Pramaan as above mentioned things we will fix in backend ?
